### PR TITLE
feat(ci): add deterministic PR intake checks

### LINKED ISSUE #3957: Add PR intake automation for roadmap alignment, size limits, and dogfooding checks
## What would you like to be added?

I would like to add a dedicated PR intake automation path that runs before or alongside normal review. The goal is to catch process-level PR risks early and route PRs more clearly before maintainers spend review time on them.

The automation should evaluate at least the following:

- whether the PR is directionally aligned with the project roadmap or should be marked for discussion;
- whether the PR is too large to review safely, with an initial hard guard around PRs above 2,000 changed lines;
- whether large features are split into a planning PR plus smaller implementation PRs;
- whether internally authored feature PRs include a dogfooding section that explains how to validate the change from a user perspective;
- whether the PR should receive labels such as roadmap-related labels, `TBD`, `welcome-pr`, or process/status labels that already exist in the repository.

This should be separate from the regular code-review bot. Code review should focus on correctness and maintainability after the PR is considered worth reviewing; PR intake should answer whether the PR is ready and directionally appropriate for review.

## Why is this needed?

The current PR flow can spend reviewer and CI resources on changes that are hard to evaluate, too large to review efficiently, or not yet aligned with the product direction. The main bottleneck is not always whether a PR can be implemented, but whether it should enter the review and merge path in its current form.

Early intake automation would help:

- reduce review load from oversized PRs;
- encourage smaller PRs that are easier to test and merge;
- make roadmap alignment visible before detailed code review;
- give authors actionable feedback before multiple review rounds;
- connect dogfooding notes to later QA and release/operation materials.

This is related to CI throughput but should not be part of the CI runtime optimization work in #3943. CI runtime optimization makes checks faster; this issue focuses on reducing avoidable review and automation churn before the expensive parts of the process.

## Additional context

Suggested initial scope:

- implement a lightweight GitHub Actions check for mechanical rules first, especially changed-line count and missing dogfooding sections;
- keep roadmap/product-direction analysis as a separate Qwen-based job or bot prompt with conservative behavior;
- only enforce dogfooding requirements for internal collaborators initially;
- comment with clear remediation steps instead of making broad code-review comments;
- use existing labels only.

Suggested labels:

- `type/feature-request`
- `category/development`
- `scope/github-actions`
- `scope/ci-cd`
- `roadmap/background-automation`

### PR DESCRIPTION
**Pull request by yiliang114**
## What this PR does

This PR adds a deterministic intake step to the existing Qwen triage job before AI triage runs.

The intake step:

- requires internally authored `feat:` PRs to provide a concrete user-perspective test plan and real evidence;
- allows up to 2,000 changed lines;
- requires a planning issue and a cannot-split rationale above 2,000 lines;
- blocks incomplete intake with an actionable, duplicate-safe comment;
- labels a fully documented oversized PR as `TBD` for a maintainer decision.

PR `edited` and `synchronize` events re-run only the mechanical intake check. They use a separate concurrency group, so they neither cancel nor repeat an in-progress AI triage run. The workflow fails closed when PR metadata is unavailable and only updates intake comments created by the repository bot.

This does not add a workflow or job. Existing Qwen triage remains the single owner of product-direction, production-code size, and implementation-risk judgments.

## Why it's needed

Current Qwen triage already handles subjective product and code-risk decisions. The remaining gap from #3957 is a small pre-AI gate for requirements that can be checked deterministically: internal dogfooding evidence and the 2,000-line planning fields.

Keeping those checks mechanical avoids spending AI and maintainer time on incomplete intake while preserving the existing triage policy as the single source of truth.

### How to verify

1. Run `node --test .github/scripts/*.test.mjs` and confirm all 70 tests pass.
2. Run `npx vitest run --config ./scripts/tests/vitest.config.ts scripts/tests/qwen-triage-workflow.test.js` and confirm all 13 tests pass.
3. Run actionlint, yamllint, Prettier, and `git diff --check`.
4. Verify the boundary cases: 2,000 changed lines pass; 2,001 lines require both planning fields and then route to `TBD`.

### Evidence (Before & After)

Before: internal feature PRs and PRs above 2,000 changed lines could enter AI triage without deterministic template validation.

After: the mechanical gate runs first, gives direct remediation for incomplete fields, and routes documented oversized PRs to a maintainer without starting another AI triage pass.

## Risk & Scope

- Main tradeoff: the 2,000-line intake threshold counts all additions and deletions as review volume; the existing triage skill separately evaluates production-code risk.
- `TBD` removal remains a maintainer action; automation only adds it.
- The canonical `pull_request_target` behavior can only be exercised after the workflow is present on the base branch.
- No new workflow, job, dependency, or AI policy is introduced.

## Linked Issues

Resolves #3957

<details>
<summary>中文说明</summary>

这个 PR 只补充 #3957 当前仍缺少的机械 intake 检查：

- 内部作者的 `feat:` PR 必须填写真实的用户视角验证步骤和证据；
- 变更超过 2,000 行时，必须填写 planning issue 和无法拆分的原因；
- 信息完整的超大 PR 标记为 `TBD`，交给 maintainer 判断。

改动继续放在现有 `Qwen Triage` workflow 和 `triage` job 内，不新增 workflow 或 job。PR 编辑和代码同步只重新运行机械检查，不取消或重复运行 AI triage。现有 triage skill 仍是产品方向、生产代码规模和实现风险判断的唯一来源。

</details>

### CODE REVIEW

File: .github/scripts/pr-intake.test.mjs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+test('allows small PR types that do not require an issue', () => {
+  assert.equal(assessPullRequestIntake(pr()).decision, 'allow');
+});
+
+test('requires fix and feat PRs to link an issue', () => {
+  for (const title of ['fix: handle empty input', 'feat(cli)!: add mode']) {
+    const result = assessPullRequestIntake(pr({ title }));
+    assert.equal(result.decision, 'block');
+    assert.ok(result.reason_codes.includes('missing_linked_issue'));
+  }
```
Comment (doudouOUC) [inline .github/scripts/pr-intake.test.mjs]: **[Suggestion]** No positive test for `fix:` PRs with a valid closing keyword. The `fix:` type is only tested on its block path (no linked issue). If `isType(title, 'fix')` were inadvertently broken (e.g., a regex change that stopped matching `fix:`), no test would catch it — a `fix:` PR with a linked issue would silently `allow` instead of `block`, and the current `missing_linked_issue` test would also stop firing, making the breakage invisible. The `feat:` happy path is covered (`'allows internal features with a real test plan and evidence'`), but `fix:` is not.

```suggestion
for (const title of ['fix: handle empty input', 'feat(cli)!: add mode']) {
    const result = assessPullRequestIntake(pr({ title }));
    assert.equal(result.decision, 'block');
    assert.ok(result.reason_codes.includes('missing_linked_issue'));
  }

  // Positive test: fix: with a valid closing keyword should allow
  {
    const result = assessPullRequestIntake(pr({ title: 'fix: null guard', body: 'Fixes #123' }));
    assert.equal(result.decision, 'allow');
  }
```

_— qwen3.7-max via Qwen Code /review_

File: .github/workflows/qwen-triage.yml [hunk 2]
```diff
@@ -259,8 +259,79 @@ jobs:
...
+              --jq '.[] | select((.body // "") | contains("<!-- qwen-pr-intake:needs-discussion -->")) | .id' |
+              tail -n 1
+          )"
+
+          case "$decision" in
+            allow)
+              if [ -n "$prior_discussion" ]; then
+                gh pr edit "$NUMBER" --repo "$GITHUB_REPOSITORY" --remove-label 'TBD' >/dev/null 2>&1 || true
+              fi
+              ;;
```
Comment (doudouOUC) [inline .github/workflows/qwen-triage.yml]: **[Suggestion]** Stale intake comments are never cleaned up across state transitions. The `allow` branch removes the TBD label but does not delete prior `<!-- qwen-pr-intake:block -->` or `<!-- qwen-pr-intake:needs-discussion -->` comments. The `block` and `needs_discussion` branches each upsert their own marker but never remove the other branch's prior comment either. A PR that transitions between states accumulates contradictory bot comments. **Failure scenario:** A `feat:` PR without a linked issue gets `block` with an "intake stopped" comment. The author adds `Resolves #123`. Intake returns `allow`. Triage runs and posts its review — but the stale block comment remains visible alongside the triage review, misleading readers about the PR's current status.

**Suggested fix:** In each branch, before upserting, query for the other markers' comment IDs and delete them:
```bash
# In allow branch, after TBD removal:
for marker in '<!-- qwen-pr-intake:block -->' '<!-- qwen-pr-intake:needs-discussion -->'; do
  old_id="$(gh api "repos/${GITHUB_REPOSITORY}/issues/${NUMBER}/comments" --paginate \
    --jq ".[] | select((.body // \"\") | contains(\"${marker}\")) | .id" | tail -n 1)"
  [ -n "$old_id" ] && gh api -X DELETE "repos/${GITHUB_REPOSITORY}/issues/comments/${old_id}" >/dev/null 2>&1 || true
done
```
Apply the same cleanup symmetrically in the `block` and `needs_discussion` branches.

_— qwen3.7-max via Qwen Code /review_

**Review by doudouOUC — COMMENTED** [review]
Reviewed. Suggestions are inline. 1 Suggestion-level finding(s) could not be anchored to the diff; see the terminal output.

_— qwen3.7-max via Qwen Code /review_
